Support ResTable staged alias type

jselbo opened the pull request:
### Overview

This non-fatal warning is printed when loading resources.asrc from the Android 13 jar: `[WARN] Unknown chunk type '200'` from:
https://github.com/robolectric/robolectric/blob/6899b8cedbe6c89cfd8723983e88d0d52a992ab9/resources/src/main/java/org/robolectric/res/android/LoadedArsc.java#L752-L754

The warning message logic is actually wrong -- it should print `child_chunk.type()` not `chunk.type()` based on the `switch` statement (though this is a bug in the original `LoadedArsc.cpp`). The `child_chunk.type()` value is `0x206` and corresponds to `RES_TABLE_STAGED_ALIAS_TYPE`.

### Proposed Changes

Add support for the new staged alias type, used by aapt2 to map pre-release resource IDs to IDs set in the final release. This is essentially a transliteration of the modifications to `libs/androidfw` in https://cs.android.com/android/_/android/platform/frameworks/base/+/2fedba9a32d9e92344eaf6e9faf5b43e1bc2ae70

Tested by running a test with latest SDK 33:
```
val resources = ApplicationProvider.getApplicationContext<Context>().resources
// pre-release resource ID
println(resources. getResourceEntryName(31391775)) // "knownActivityEmbeddingCerts"
```
Can also dump `mAliasId` map -- there are 58 entries whose keys are pre-release SDK 33 resource IDs.
Also no more warning printed.

Unrelated - I believe there is a bug in the existing `ResTable_lib_entry` logic where the first entry should be constructed with offset after the header size, and the iteration end condition should include `* ResTable_lib_entry.SIZEOF` similarly. I wasn't sure how to test this path.

utzcoz commented:
@jselbo Thanks for your contribution. Please follow https://github.com/robolectric/robolectric/wiki/Robolectric's-code-style to style your changes and make your building passed.

jselbo commented:
@utzcoz I'm happy to, though it looks like these files are not formatted to begin with. If I run the `google-java-format-diff.py` script to format the changed lines, amend, and repeat, the formatter still continues to produce changes after several cycles.

Are you comfortable running google-java-format over the entire files here?

utzcoz commented:
> Are you comfortable running google-java-format over the entire files here?

We don't expect full-file formatting for all files, because there are so many files have left formatting problems, and full formatting will affect many existing or upcoming changes because of massive conflict. We need every PR's changes pass diff-changes formatting checking, and the full  project will become formatting friendly after some months or years.

>  the formatter still continues to produce changes after several cycles.

It's recommended to format your code when you think your PR is ready for reviewing. You can wrap command to format Java code changes to a script, and run it quickly before you every pushing. It's convenient and easy.

jselbo commented:
Just to clarify: the "Check Code Formatting" CI job runs this script: https://raw.githubusercontent.com/google/google-java-format/v1.15.0/scripts/google-java-format-diff.py  which basically pipes line numbers reported by `git diff` to `google-java-format --lines`, but the lines reported by `git diff` span outside the lines I actually touched. Running the formatter changes more lines, and running the script subsequent times continues to produce changes due to the ever-expanding scope of the `git diff` output. Running the script over and over until success exit code would ultimately format the entire files.

I think the best solution for my PR here is to format the files in a commit prior to the new feature. Let me update.

jselbo commented:
Sorry about the churn - should be good to review now.

utzcoz commented on resources/src/main/java/org/robolectric/res/android/CppAssetManager2.java:
```diff
@@ -259,32 +260,32 @@ static class FindEntryResult {
   }
 
-//  AssetManager2() { memset(&configuration_, 0, sizeof(configuration_)); }
+  //  AssetManager2() { memset(&configuration_, 0, sizeof(configuration_)); }
```
Are these formatting changes generated by google-java-format? I remember it will not report error for these existing comments.

jselbo replied:
Yes, changed by the formatter

utzcoz commented on resources/src/main/java/org/robolectric/res/android/CppAssetManager2.java:
```diff
@@ -722,8 +732,8 @@ private ApkAssetsCookie FindEntry(int resid, short density_override,
-            if ((best_config == null || this_config.isBetterThan(best_config, desired_config)) ||
-                (package_is_overlay && this_config.compare(best_config) == 0)) {
+            if ((best_config == null || this_config.isBetterThan(best_config, desired_config))
+                || (package_is_overlay && this_config.compare(best_config) == 0)) {
```
Are these formatting changes generated by google-java-format? If not, please keep the origin formatting if you don't change these lines.

jselbo replied:
Yes, this is from google-java-format

I split the formatting changes into a separate commit in this PR to easily distinguish between the formatting changes and my new changes

utzcoz replied:
Thanks for your confirmation. I will double-check it locally again.

utzcoz commented:
> but the lines reported by git diff span outside the lines I actually touched. 

@jselbo You can check https://github.com/robolectric/robolectric/wiki/Running-google-java-format for potential reason that I found google-java-format formats more lines.

utzcoz commented on resources/src/main/java/org/robolectric/res/android/LoadedArsc.java:
```diff
@@ -765,6 +773,68 @@ static LoadedPackage Load(
             }
             break;
 
+          case RES_TABLE_STAGED_ALIAS_TYPE:
```
It might need a test as you posted before.

jselbo commented:
Sorry - I'm not sure why I was getting formatting changes outside of my changed lines when I originally tried. I removed the unrelated formatting changes and added a unit test. Ready for another review

hoisie commented:
Thanks @jselbo for the contribution. I am running it across some tests internally at Google to check for potential compatibility issues.

jselbo commented:
Friendly bump

hoisie reviewed (approved):
This is great, thanks!
